packages/runtime/container-runtime/src/opLifecycle/opSplitter.ts

```diff
@@ -203,8 +169,51 @@ export class OpSplitter {
 			referenceSequenceNumber: batch.referenceSequenceNumber,
 		};
 	}
+
+	public processChunk(message: ISequencedDocumentMessage): ProcessChunkResult {
+		assert(isChunkedMessage(message), "message not of type ChunkedOp");
+
+		// TODO: Verify whether this should be able to handle server-generated ops (with null clientId)
+		// eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
+		const clientId = message.clientId as string;
+		const chunkedContent = message.contents as IChunkedOp;
+		this.addChunk(clientId, chunkedContent, message);
+
+		if (chunkedContent.chunkId < chunkedContent.totalChunks) {
+			// We are processing the op in chunks but haven't reached
+			// the last chunk yet in order to reconstruct the original op
+			return {
+				isFinalChunk: false,
+			};
+		}
+
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		const serializedContent = this.chunkMap.get(clientId)!.join("");
+		this.clearPartialChunks(clientId);
+
+		const newMessage = { ...message };
+		newMessage.contents = serializedContent === "" ? undefined : JSON.parse(serializedContent);
+		newMessage.type = chunkedContent.originalType;
+		newMessage.metadata = chunkedContent.originalMetadata;
+		newMessage.compression = chunkedContent.originalCompression;
+		return {
+			message: newMessage,
+			isFinalChunk: true,
+		};
+	}
+}
+
+export interface INotFinalChunkResult {
+	readonly isFinalChunk: false;
+}
+
+export interface IFinalChunkResult {
+	readonly isFinalChunk: true;
+	readonly message: ISequencedDocumentMessage;
 }
 
+export type ProcessChunkResult = INotFinalChunkResult | IFinalChunkResult;
```

Comment:
nit: I would just inline the Final/Notfinal shapes rather than naming them, since they're not used otherwise.